src/bin/pg_send_cancellation/Makefile

```diff
@@ -10,7 +10,7 @@ PG_LDFLAGS += $(LDFLAGS)
 include $(citus_top_builddir)/Makefile.global
 
 # We reuse all the Citus flags (incl. security flags), but we are building a program not a shared library
-override CFLAGS := $(filter-out -shared,$(CFLAGS))
+override CFLAGS := $(filter-out -shared,$(CFLAGS)) -fno-lto
```

Comment:
Added this to build citus in almalinux:9 (el/9 packages)

Reply:
why do we need those @JelteF ?

Reply:
Could definitly use a comment. But it's because we sometimes build with a newer version of gcc than postgres was built with, which breaks LTO (link-time optimization). Disabling it can have some perf impact, but since pg_send_cancellation is only used for tests anyway, this seems fine.

Reply:
Added comment